packages/server/src/core/degree_status/postprocessing.rs

```diff
@@ -75,24 +125,46 @@ impl DegreeStatus {
         }
     }
 
-    fn medicine_preclinical_avg(&self) -> f64 {
-        let grades = self
-            .course_statuses
-            .iter()
-            .filter(|course_status| course_status.course.is_medicine_preclinical())
+    fn medicine_preclinical_avg(&self, catalog: &Catalog) -> f32 {
+        let rule_all_courses = self.course_statuses.iter().filter(|course_status| {
+            get_courses_of_rule_all(catalog)
+                .contains(&course_status.r#type.clone().unwrap_or_default())
+        });
+
+        let highest_sport_grades =
+            self.get_highest_grade_courses_up_to_credit_requirement(catalog, SPORT_BANK_NAME);
+
+        let highest_accumulated_credit_grades = self
+            .get_highest_grade_courses_up_to_credit_requirement(
+                catalog,
+                MEDICINE_ACCUMULATE_CREDIT_BANK_NAME,
+            );
+
+        let highest_grades = rule_all_courses
```

Comment:
Maybe we should rename `highest_grades` to `highest_grade_courses` to indicate that this is actually an iterator of `CourseStatus`es and not of `Grade`s